app/code/Magento/InventoryCatalogSearch/Test/Integration/Model/Indexer/FulltextTest.php

```diff
@@ -0,0 +1,309 @@
+        $this->storeCodeBefore = $this->storeManager->getStore()->getCode();
+
+        $this->productRepository = Bootstrap::getObjectManager()->get(ProductRepositoryInterface::class);
+        $this->productSku1 = $this->productRepository->get('SKU-1');
+        $this->productSku2 = $this->productRepository->get('SKU-2');
+        $this->productSku3 = $this->productRepository->get('SKU-3');
+        $this->productAction = Bootstrap::getObjectManager()->get(Action::class);
+    }
+
+    /**
+     * @param string $store
+     * @param array $expectedSearchSize
+     * @return void
+     *
+     * @dataProvider reindexAllDataProvider
+     */
+    public function testReindexAll(string $store, array $expectedSearchSize)
+    {
+        $this->storeManager->setCurrentStore($store);
+        $this->indexer->reindexAll();
+
+        $products = $this->search('Orange');
+        $this->assertCount($expectedSearchSize[0], $products);
+        if ($expectedSearchSize[0] > 0) {
+            $this->assertEquals($this->productSku1->getId(), $products[0]->getId());
+        }
+
+        $products = $this->search('Simple Product');
+        $this->assertCount($expectedSearchSize[1], $products);
+    }
+
+    /**
+     * @return array
+     */
+    public function reindexAllDataProvider(): array
+    {
+        return [
+            ['store_for_eu_website', [1, 1]],
+            ['store_for_us_website', [0, 1]],
+            ['store_for_global_website', [1, 2]],
+        ];
+    }
+
+    /**
+     * @param string $store
+     * @param array $expectedSearchSize
+     * @return void
+     *
+     * @dataProvider reindexRowDataProvider
+     */
+    public function testReindexRowAfterEdit(string $store, array $expectedSearchSize)
+    {
+        $this->storeManager->setCurrentStore($store);
+
+        $this->indexer->reindexAll();
+
+        $this->productSku1->setName('Simple Product Red');
+        $this->productRepository->save($this->productSku1);
+
+        $products = $this->search('Orange');
+        $this->assertCount(0, $products);
+
+        $products = $this->search('Red');
+        $this->assertCount($expectedSearchSize[0], $products);
+        if ($expectedSearchSize[0] > 0) {
+            $this->assertEquals($this->productSku1->getId(), $products[0]->getId());
+        }
+        $products = $this->search('Simple Product');
+        $this->assertCount($expectedSearchSize[1], $products);
+    }
+
+    /**
+     * @return array
+     */
+    public function reindexRowDataProvider(): array
+    {
+        return [
+            ['store_for_eu_website', [1, 1]],
+            ['store_for_us_website', [0, 1]],
+            ['store_for_global_website', [1, 2]],
+        ];
+    }
+
+    /**
+     * @param string $store
+     * @return void
+     *
+     * @dataProvider reindexAllDataProvider
+     */
+    public function testReindexRowAfterMassAction(string $store)
```

Comment:
What is `massAction` in index context?
Moreover, we call `reindexAll` inside of tests

Pls, check all names in testcase

Reply:
We call reindex, then we updating name for two products:
```
$productIds = [
            $this->productSku1->getId(),
            $this->productSku2->getId(),
        ];
        $attrData = [
            'name' => 'Simple Product Common',
        ];

        $this->productAction->updateAttributes($productIds, $attrData, $this->storeManager->getStore()->getId());
```